### DISCUSSION TITLE
Board dependent configuration file?

### DISCUSSION THREAD

**Discussion by OOHehir** [LIFsCode/ELOC-3.0#39] [Ideas]
@EDsteve @LIFsCode this is a suggested change which I think may help with different boards, testing & general code improvement. 

I've done a bit of testing using the ESP32-WROVER-IE-N16R8  board as it allows debugging using a JTAG. Using the JTAG on the the ELOC board was unsuccessful as the pins were in use for peripherals. Changing the code from board to another is cumbersome as the pin definitions have to be changed, peripherals disabled etc. Also I found it difficult to centralise certain build options (such as a default volume for recording, the bits per sample for a particular microphone, enabling/ disabling certain AI related values etc etc). The file config.h seems a reasonable location for these settings but it's not possible to link currently from a file in /lib. The reason is that config.h includes other files which then causes compile errors (relative path issue).

I would like to suggest that a single configuration header file is used to set available peripherals, pin outs, microphone settings & project configurations, for example this [file](https://github.com/LIFsCode/ELOC-3.0/blob/ai/eloc610LowPowerPartition/src/project_config.h).  This is just splitting off a lot of the definitions that were already in config.h into a separate/ child file.

The idea, as per the comment in the file is that the file platform.ini would define a BOARD & that would determine all the subsequent pinouts, peripherals available, etc. Numerous tasks would be available in VSC that could build all the different boards but all with the same code.

I don't think it would be too difficult to implement but please let me know your thoughts & if you think its a waste of time.

Regards,

Owen

**Comment by LIFsCode** [comment]
@OOHehir : Great idea. Thanks. The config.h is a bit of a mess same as main.cpp. So this will definetly make things more clear.

@EDsteve : could we get rid of the SPI version? I don't think the current code would run onto it because of several missing HW peripherals and I don't think it would be worth the effort to backport everything to the ELOC 2.7 HW unless you have 100ds of boards of the old HW which are required to be used. Decission is up to you.

**Comment by EDsteve** [comment]
@LIFsCode @OOHehir 
There are only 30 or now only 25 ELOC2.7 (SPI) out there. And they are aware that there will be no further firmware updates for it. So we can purely concentrate on the new ELOC3.0 upwords.

**Comment by OOHehir** [comment]
https://github.com/LIFsCode/ELOC-3.0/commit/8b1927dd87c986fe032915f80c14a8fb6e63e5d9
https://github.com/LIFsCode/ELOC-3.0/commit/0f461d38e3ff9f31bd74572d8fb05547dc3a53b0
